### PR TITLE
[low-power] fix CSL_RECEIVER build option

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR removes the config setting `OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE`automatically when it's 1.2 and MTD. The reason is that it's in core config and it's also used in platform(radio) which cannot access core config. 

If `CSL_RECEIVER` is not set in options in cmake and it's 1.2, then in core config, `OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE` would be turned on if it's MTD. While in platform, it's not turned on. And then there would be a link issue.

### CODE REVIEW

File: src/core/config/mac.h [hunk 1]
```diff
@@ -314,18 +314,6 @@
...
- * This setting configures the CSL receiver feature in Thread 1.2.
- * This is compulsory for 1.2 MTD, optional for 1.2 FTD.
- *
- */
-#ifndef OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-#define OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE \
-    (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2) && OPENTHREAD_MTD
-#endif
-
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
```
Comment (simonlingoogle) [inline src/core/config/mac.h:317]: seems to have many warnings?

**Review by simonlingoogle — COMMENTED** [review]
We should catch the error with compile errors rather than leave it to link error?

File: src/core/config/mac.h (hunk 1 above)
Comment (Irving-cl) [reply src/core/config/mac.h:329]: Has defined it as 0 to avoid warnings.

File: src/core/config/mac.h [hunk 2]
```diff
@@ -318,12 +318,12 @@
  * @def OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
  *
  * This setting configures the CSL receiver feature in Thread 1.2.
- * This is compulsory for 1.2 MTD, optional for 1.2 FTD.
+ * Set it to 0 by default because it is also used in 'platform' which cannot access the core config. To enable this
+ * option, it has to be set in build options when building.
```
Comment (simonlingoogle) [inline src/core/config/mac.h:322]:
```suggestion
```
I think users already know they need to specify build options when building.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/core/config/mac.h (hunk 2 above)
Comment (Irving-cl) [reply src/core/config/mac.h]: Make sense. Has removed it.

**Review by librasungirl — APPROVED** [review]

**Pull request merged**